### PR TITLE
Ignore opaque or counted resources from Pod QoS

### LINKED ISSUE #33108: Pod QOS calculation is confused by opaque resources
**Scenario:**

A pod that makes a request for a GPU, but makes no cpu or memory request.

**Expected result:**

Pod has BestEffort QOS

**Actual result:**

Pod has Burstable QOS

**Details:**

The QOS classification should align with resources that are cgroup managed on the node.  We should ignore opaque resources that are not node enforced.

See: https://github.com/kubernetes/kubernetes/issues/32468#issuecomment-246875226

FYI: @vishh @ConnorDoyle @sjenning @kubernetes/sig-scheduling @kubernetes/sig-node

### PR DESCRIPTION
**Pull request by vishh**
Fixes #33108

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/33112)

### CODE REVIEW

**Comment by vishh** [comment]
cc @ConnorDoyle

**Comment by derekwaynecarr** [comment]
@vishh -- you were quick to code!  change and test cases LGTM.

**Comment by vishh** [comment]
@derekwaynecarr Haven't gotten much time these days to write code. So used this issue as a quick opportunity to post an easy PR 😉

File: pkg/kubelet/qos/qos.go [hunk 1]
```diff
@@ -118,17 +124,14 @@ func GetQOS(container *api.Container) QOSList {
 	return resourceToQOS
 }
 
-// supportedComputeResources is the list of supported compute resources
-var supportedComputeResources = []api.ResourceName{
-	api.ResourceCPU,
-	api.ResourceMemory,
-}
+// supportedComputeResources is the list of compute resources for with QoS is supported.
```
Comment (ConnorDoyle) [inline pkg/kubelet/qos/qos.go]: s/for with/for which

**Review by ConnorDoyle — APPROVED** [review]
Nice minimal diff, tests LGTM

**Pull request merged**